pkg/chunk/series_store.go

```diff
@@ -291,3 +297,92 @@ func (c *seriesStore) lookupChunksBySeries(ctx context.Context, from, through mo
 	result, err := c.parseIndexEntries(ctx, entries, nil)
 	return result, err
 }
+
+// Put implements ChunkStore
+func (c *seriesStore) Put(ctx context.Context, chunks []Chunk) error {
+	for _, chunk := range chunks {
+		if err := c.PutOne(ctx, chunk.From, chunk.Through, chunk); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// PutOne implements ChunkStore
+func (c *seriesStore) PutOne(ctx context.Context, from, through model.Time, chunk Chunk) error {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Horribly, PutChunks mutates the chunk by setting its checksum.  By putting
+	// the chunk in a slice we are in fact passing by reference, so below we
+	// need to make sure we pick the chunk back out the slice.
+	chunks := []Chunk{chunk}
+
+	err = c.storage.PutChunks(ctx, chunks)
+	if err != nil {
+		return err
+	}
```

Comment:
This reuse overcomplicates this; I'd put the calls to c.calculateSeriesEntries here.